src/CMake/FindZlib.cmake

```diff
@@ -43,15 +43,30 @@ if (WIN32)
         #set(ZLIB_LIBRARY "${ZLIB_LIB}" CACHE STRING "zlib library" FORCE)
         set(ZLIB_LIBRARY "${ZLIB_LIBRARY_DIR}/${ZLIB_LIB}" CACHE STRING "full path to zlib library" FORCE)
     endif ()
-else(WIN32)
-    if(VISIT_ZLIB_DIR)
-        # We've told VisIt where to look for zlib.
-        SET_UP_THIRD_PARTY(ZLIB lib include z)
-        if(ZLIB_FOUND)
-            # use full path here, instead of just lib file.
+else (WIN32)
+    # Have we told VisIt where to look for zlib?
+    if (VISIT_ZLIB_DIR)
+        if (LINUX AND CMAKE_SIZEOF_VOID_P EQUAL 8 AND EXISTS ${ZLIB_DIR}/lib64)
+            SET_UP_THIRD_PARTY(ZLIB lib64 include z)
+        else ()
+            SET_UP_THIRD_PARTY(ZLIB lib include z)
+        endif ()
+        if (ZLIB_FOUND)
+        # use full path here, instead of just lib file.
             set(ZLIB_LIBRARY "${ZLIB_LIBRARY_DIR}/${ZLIB_LIB}" CACHE STRING "zlib library" FORCE)
-        endif()
-    endif()
+        endif ()
+    else ()
+        # message(STATUS "ZLIB_LIB_NAME = ${ZLIB_LIB_NAME}.")
+        if ((NOT ZLIB_LIB_NAME) OR ZLIB_LIB_NAME STREQUAL "")
```

Comment:
We require visit to use the zlib we build. build visit modules were updated accordingly, eg vtk is now built with 'system' zlib pointing to our build.  This was done to reduce the incompatibilities between various system zlibs, third party-built zlibs and visit-built zlib.

Reply:
I hear you, but then I do not know how to build my project, which includes visit, consistently.  We want one zlib that is specified outside of VisIt, then used to build VisIt, our composer, etc.
I can take this out of the pull request and keep it in only our branch, but does it harm the project to allow this (external specification of zlib) as an option?

Reply:
Can you set VISIT_ZLIB_DIR to point to your zlib?

Reply:
Yes, I can.  I thought it was nice to have a fallback, but I will do this.

Reply:
BTW, when I do that, -DVISIT_ZLIB_DIR=/usr, I still need the code for determining lib64 versus lib.  Is that okay?

Reply:
That sounds fine, though it may be simpler to just check existence of ${ZLIB_DIR}/lib64.
Other packages handle it this way.
Think I will enter a ticket to have our SET_UP_THIRD_PARTY handle lib64 locations, should make things easier in the future.